### PR TITLE
Add new lint macro_no_longer_exported

### PR DESCRIPTION
**Pull request by malenaohl**
Addresses checkbox in https://github.com/obi1kenobi/cargo-semver-checks/issues/946

### CODE REVIEW

File: src/lints/macro_no_longer_exported.ron [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                }
+            }
+            current {
+                item {
+                    ... on Macro {
+                        name @filter(op: "=", value: ["%name"])
+                        # Check the macro still exists but is no longer public API
+                        # and isn't doc(hidden) (which would be caught by another lint)
+                        public_api_eligible @filter(op: "=", value: ["$false"])
+                        doc_hidden @filter(op: "=", value: ["$false"])
```
Comment (obi1kenobi) [inline src/lints/macro_no_longer_exported.ron:26]:
```suggestion
                        doc_hidden @filter(op: "!=", value: ["$true"])
```

File: src/lints/macro_no_longer_exported.ron [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                            begin_line @output
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "true": true,
+        "false": false,
```
Comment (obi1kenobi) [inline src/lints/macro_no_longer_exported.ron:39]:
```suggestion
```

File: src/lints/macro_no_longer_exported.ron [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    error_message: "A macro that was previously exported is no longer exported. This breaks downstream code that used the macro.",
```
Comment (obi1kenobi) [inline src/lints/macro_no_longer_exported.ron:41]:
```suggestion
    error_message: "A macro that was previously exported with #[macro_export] is no longer exported. This breaks downstream code that used the macro.",
```

**Review by obi1kenobi — COMMENTED** [review]
Small tweaks.

File: src/lints/macro_no_longer_exported.ron [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Macro {
+                        name @filter(op: "=", value: ["%name"])
+                        # Check the macro still exists but is no longer public API
+                        # and isn't doc(hidden) (which would be caught by another lint)
+                        public_api_eligible @filter(op: "=", value: ["$false"])
```
Comment (malenaohl) [inline src/lints/macro_no_longer_exported.ron:25]:
```suggestion
                        public_api_eligible @filter(op: "!=", value: ["$true"])
```

**Pull request merged**